app/src/main/java/com/example/drawingactivity/DialogUtils.kt

```diff
@@ -0,0 +1,169 @@
+                when (textField.text.toString().trim()) {
+                    "", textField.hint.toString() -> {
+                        onSubmit.invoke(textField.hint.toString())
+                    }
+                    else -> {
+                        onSubmit.invoke(textField.text.toString().trim())
+                    }
+                }
+                onSubmit.invoke(null)
+            }
+            setNegativeButton(context.getString(R.string.addLayerNegativeButton)) { _, _ ->
+                onSubmit.invoke(null)
+            }
+            setOnCancelListener {
+                onSubmit.invoke(null)
+            }
+        }
+        val dialog: AlertDialog = builder.create()
+        dialog.show()
+        val submitButton: Button = dialog.getButton(AlertDialog.BUTTON_POSITIVE)
+        textField.addTextChangedListener {
+            onEditText.invoke(it.toString().trim())
+            val errorStatus = checkForErrors(context, layerList, it.toString().trim())
+            if (errorStatus != null) {
+                errorView.text = errorStatus
+                errorView.isVisible = true
+                submitButton.isEnabled = false
+            } else errorView.isVisible = false
+        }
+    }
+
+    fun layerListBottomSheet(
+        context: Context,
+        layerList: List<String>,
+        /**
+         * Opens a Dialog to Delete an object in the list.
+         */
+        onDelete: ((Int) -> Unit)?,
+        /**
+         * Opens a Dialog to Edit an object in the list.
+         */
+        onEdit: ((Int?) -> Unit)?,
+        /**
+         * Called when the Dialog Completes through any listener.
+         * This will reset all values to null in preparation for the next opening of the Dialog.
+         */
+        onDialogComplete: (() -> Unit)?,
+    ): ((Int, String?, String) -> Unit) {
+        val sheet = BottomSheetDialog(context)
+        val view = LayoutInflater.from(context).inflate(R.layout.recycler_layer_list_layout, null)
+        val data = layerList.map { LayerViewModel(it) }.toMutableList()
+        val recyclerView = view.findViewById<RecyclerView>(R.id.listOfLayers)
+        recyclerView.apply {
+            layoutManager = LinearLayoutManager(context)
+            adapter = RecyclerViewAdapter(
+                data = data,
+                onDeleteLayer = { onDelete?.invoke(it) },
+                onEditName = { onEdit?.invoke(it) }
+            )
+        }
+        sheet.apply {
+            setOnDismissListener {
+                dismiss()
+                onDialogComplete?.invoke()
+            }
+            setContentView(view)
+            show()
+        }
+        val behavior = BottomSheetBehavior.from(view.parent as View)
+        behavior.state = BottomSheetBehavior.STATE_EXPANDED
+        behavior.isDraggable = false
+        return { index, newName, operation ->
+            when (operation) {
+                "EDIT" -> {
```

Comment:
We shouldn't be hardcoding things without constants. And this looks like what you'd use an enum for